### PR TITLE
bpo-44693: Update __future__ entry in Doc/glossary.rst

### PR DESCRIPTION
**Pull request by StevenHsuYL**
Replace sentence with confusing "pseudo-module" with two sentences separating future statements and the __future__ module.

https://bugs.python.org/issue44693

### CODE REVIEW

File: Doc/glossary.rst [hunk 1]
```diff
@@ -451,8 +451,10 @@ Glossary
       for best practices on working with annotations.
 
    __future__
-      A pseudo-module which programmers can use to enable new language features
-      which are not compatible with the current interpreter.
+      A :ref:`future statement <future>`, "from __future__ import *feature*
+      ...", directs the compiler to compile the current module using syntax or
```
Comment (merwok) [inline Doc/glossary.rst]: Code can use three backticks ` rather than quote marks

Comment (terryjreedy) [reply Doc/glossary.rst]: I will look at the result either way.  I will also move the paragraph break and remove the duplicate module reference that follows.

Comment (terryjreedy) [reply Doc/glossary.rst]: Two backticks, not one or three, are needed for inline code in .rst.  I decided to remove ` ...` from the quoted code; one interested in the full syntax will have to look at the doc.  I also removed the paragraph break.

Comment (terryjreedy) [reply Doc/glossary.rst]: I changed `*feature*` to `<feature>` as the latter is, I believe, used more for a syntax variable.

**Pull request merged**